pkg/client/factory.go

```diff
@@ -207,9 +210,19 @@ func (s *SingletonClients) ResourceInfo(obj *unstructured.Unstructured, retryOnN
 		return nil, err
```

Comment:
For above `getResourceMapping`, just want to clarify due to my knowledge missing, why the condition is when gvk is empty, we return mapping, but not when it's not empty? @adityabhatia  
```
	if gvk.Empty() {
		return mapping, nil
	}
``

Reply:
if the gvk is empty, there is no way that a mapping can be resolved since it is missing type information. Thus no rest mapper would be able to resolve the type here. Instead this will cause the resource to stay unstructured without a mapping and then later on fail in the concurrent apply (which is fine since only the failing resource should not be applied)